// type-json-mapper/esm/lib/utils.js
import 'reflect-metadata';
export var META_KEY = '__TRANSKEY__';
export var getJsonProperty = function (target, propertyKey) {
    return typeof Reflect.getMetadata === 'function' && Reflect.getMetadata(META_KEY, target, propertyKey);
};
export var setProperty = function (value) {
    return typeof Reflect.metadata === 'function'
        ? Reflect.metadata(META_KEY, value)
        : function (target, propertyKey) {
            console.warn('Reflect.metadata 未正确加载。');
        };
};
export var isObject = function (target) { return Object.prototype.toString.call(target) === '[object Object]'; };
export var isArray = function (target) { return Object.prototype.toString.call(target) === '[object Array]'; };
export var hasAnyNullOrUndefined = function () {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    return args.some(function (arg) { return arg === null || arg === undefined; });
};
export var isNotBasicType = function (target) { return ['string', 'number', 'boolean'].every(function (type) { return type !== typeof target; }); };
export var isInvalidDate = function (date) { return date instanceof Date && isNaN(date.getTime()); };
export var formatDate = function (timestamp, format) {
    if (format === void 0) { format = 'Y-M-D h:m:s'; }
    var date = new Date(timestamp);
    if (isInvalidDate(date)) {
        return timestamp;
    }
    var formatNumber = function (n) { return (n >= 10 ? "" + n : "0" + n); };
    return format
        .replace('Y', "" + date.getFullYear())
        .replace('M', formatNumber(date.getMonth() + 1))
        .replace('D', formatNumber(date.getDate()))
        .replace('h', formatNumber(date.getHours()))
        .replace('m', formatNumber(date.getMinutes()))
        .replace('s', formatNumber(date.getSeconds()));
};
export var getRandomInt = function (min, max) {
    if (!min || !max) {
        return 0;
    }
    var range = max - min;
    var rand = Math.random();
    return min + Math.round(rand * range);
};
export var getRandomString = function (length, chars) {
    if (chars === void 0) { chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz1234567890'; }
    return !length ? '' : new Array(length).fill(0).reduce(function (res) { return "" + res + chars.charAt(Math.floor(Math.random() * chars.length)); }, '');
};
export var getRandomFloat = function (length) {
    if (!length) {
        return 0;
    }
    var decimal = getRandomInt(1, length - 2);
    var numStr = getRandomString(length, '123456789');
    var floatStr = parseFloat(numStr.split('').reduce(function (res, char, index) { return "" + res + (length - index === decimal ? '.' : '') + char; }, ''));
    return floatStr;
};
